rust/kernel/sync/arc.rs

```diff
@@ -475,6 +512,44 @@ impl<T> UniqueArc<MaybeUninit<T>> {
     /// Converts a `UniqueArc<MaybeUninit<T>>` into a `UniqueArc<T>` by writing a value into it.
     pub fn write(mut self, value: T) -> UniqueArc<T> {
         self.deref_mut().write(value);
+        // SAFETY: we just initialized `self`
+        unsafe { self.assume_init() }
+    }
+
+    /// Initialize the value after creating the `UniqueArc`.
+    #[allow(clippy::type_complexity)]
+    pub fn pin_init_now<E>(
+        mut self,
+        init: impl PinInit<T, E>,
+    ) -> core::result::Result<Pin<UniqueArc<T>>, (E, UniqueArc<MaybeUninit<T>>)> {
```

Comment:
Would clippy still warn if this was `Result<Pin<UniqueArc<T>>, (E, Self)>`?

Reply:
Nope 👍